File: 976-largest-perimeter-triangle/976-largest-perimeter-triangle.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        for i in reversed(range(2,len(nums))):
            if nums[i-2] + nums[i-1] > nums[i]:
                return nums[i-2] + nums[i-1] + nums[i]
        return 0

File: 976-largest-perimeter-triangle/976-largest-perimeter-triangle.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var largestPerimeter = function(nums) {
    nums.sort((a, b) => b - a);
    
    let [s1, s2, s3] = [nums[0], nums[1], nums[2]];
    
    for (let i = 3; i < nums.length; i++) {
        if (s1 < s2 + s3) return s1 + s2 + s3;
        [s1, s2, s3] = [s2, s3, nums[i]];
    }
    
    return s1 < s2 + s3 ? s1 + s2 + s3 : 0;
};